app/javascript/src/components/Editor.tsx

```diff
@@ -118,6 +122,40 @@ class Editor extends React.Component<EditorProps, EditorState> {
     this.model = model
   }
 
+  private handleSceneName = (
+    currentState: any,
+    newStory: any,
+    ) => {
+    let currentNodes = [...currentState.story.nodes]
+    let updatedName: string = ''
+    let latestLink = Object.assign(new DefaultLinkModel(), currentState.story.links[currentState.story.links.length - 1])
+    let sourcePort = Object.assign(new DefaultPortModel(false, ''))
+    let targetNode = Object.assign(new DefaultNodeModel())
+
+    currentNodes.forEach(node => {
+      if(node.id === latestLink.source) {
+        sourcePort = node.ports.filter((port: DefaultPortModel) => port.id === latestLink.sourcePort)
+        updatedName = sourcePort[0].label
+      }
+
+      if(node.id === latestLink.target) {
+         targetNode = node
+      }
+    })
+
+    if(targetNode.name === "New Scene") {
```

Comment:
This part feels a little weird just in case "New Scene" is not always going to be the default name. The add scene method just adds the node with a string for the title, but we could possibly extract both of them out and save them somewhere instead of typing "New Scene" all the time. Maybe?

Reply:
Yeah, if we have that string defined in some other file somewhere as well, would be a real good call to pull that into a constant that gets imported here. Odds are we'll never change that, but if we did it would be a shame to lose this functionality.

Reply:
Added a constants.ts file to export out of. It's a bit more future-proof now at least